lib/web/fetch/request.js

```diff
@@ -39,6 +37,21 @@ const requestFinalizer = new FinalizationRegistry(({ signal, abort }) => {
   signal.removeEventListener('abort', abort)
 })
 
+function validateAndNormalizeMethod (method) {
+  if (!isValidHTTPToken(method)) {
+    throw new TypeError(`'${method}' is not a valid HTTP method.`)
+  }
+
+  const upperCase = method.toUpperCase()
+
+  if (forbiddenMethodsSet.has(upperCase)) {
+    throw new TypeError(`'${method}' HTTP method is unsupported.`)
+  }
+
+  // Note: must be in uppercase
+  return normalizedMethodRecordsBase[upperCase] ?? method
+}
```

Comment:
Why dont you move it to util?
cam you write some test only for this method?

Reply:
This is covered by wpt.